core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -210,6 +219,51 @@ class BlockManagerMasterEndpoint(
     case StopBlockManagerMaster =>
       context.reply(true)
       stop()
+
```

Comment:
Can we add the overall flow of the messages here ?

Reply:
Added some comments here, please take a look. Thanks.